airflow/contrib/hooks/bigquery_hook.py

```diff
@@ -1705,16 +1710,26 @@ def get_datasets_list(self, project_id=None):
                    }
                 ]
         """
+
         dataset_project_id = project_id if project_id else self.project_id
 
-        try:
-            datasets_list = self.service.datasets().list(
-                projectId=dataset_project_id).execute(num_retries=self.num_retries)['datasets']
-            self.log.info("Datasets List: %s", datasets_list)
+        optional_params = {'all': all_datasets}
+        if max_results:
+            optional_params['maxResults'] = max_results
 
-        except HttpError as err:
-            raise AirflowException(
-                'BigQuery job failed. Error was: {}'.format(err.content))
+        request = self.service.datasets().list(
+            projectId=dataset_project_id,
+            **optional_params)
+
+        datasets_list = []
+
+        while request is not None:
+            response = request.execute(num_retries=self.num_retries)
+            datasets_list.extend(response['datasets'])
+            request = self.service.datasets().list_next(previous_request=request,
+                                                        previous_response=response)
+
+        self.log.info("Datasets List: %s", datasets_list)
```

Comment:
```suggestion
```
I haven't seen it before, but it seems to me that there will be too much information in the logs.

Reply:
Done